FMDN extension: documentation improvement regarding testing with debug device model (#15996)
nrfconnect/sdk-nrf: samples/bluetooth/fast_pair/locator_tag/README.rst

   Images in the testing section are generated for the debug device model registered by Nordic Semiconductor in the Google Nearby Console (see :ref:`fast_pair_locator_tag_google_device_model`).
   The debug device model name is covered by asterisks and the default Fast Pair logo is displayed instead of the one specified during the device model registration.

   If the test Android device uses a primary email account that is not on Google's email allow list for the FMDN feature, testing steps will fail at the FMDN provisioning stage for the default debug (uncertified) device model.
   To be able to test with debug device models, register your development email account by completing Google's device proposal form.

>    The link to the form is available in the blue banner note at the beginning of `Fast Pair Find My Device Network extension` specification.

Reviewer: as above

   Images in the testing section are generated for the debug device model registered by Nordic Semiconductor in the Google Nearby Console (see :ref:`fast_pair_locator_tag_google_device_model`).
   The debug device model name is covered by asterisks and the default Fast Pair logo is displayed instead of the one specified during the device model registration.

>    If the test Android device is set up for the primary email account that is not part of the Google email allow list for the FMDN feature, the following testing steps will fail at the FMDN provisioning stage for the debug (uncertified) device model model used by default in this sample.

Reviewer: ```suggestion
   If the test Android device uses a primary email account that is not on Google's email allow list for the FMDN feature, testing steps will fail at the FMDN provisioning stage for the default debug (uncertified) device model.
```

>    If the test Android device is set up for the primary email account that is not part of the Google email allow list for the FMDN feature, the following testing steps will fail at the FMDN provisioning stage for the debug (uncertified) device model model used by default in this sample.

Reviewer 2: Can we move it to requirements section (user should be aware of it early I think) and rephrase it a bit? Something like:
```
By default the sample uses a debug Fast Pair Model ID.
If you want to test FMDN with a debug Model ID, primary email account used by your Android phone must be allowlisted by Google.
Use `Fast Pair Device Proposal Form`_ from Google to register your development email for the allow list.
```

Author: this is up to @annwoj, @divipillai. I can move it there

Reviewer 2: I think it may be reasonable to let user know about the limitation early to avoid disappointment on test start

Reviewer 3: I think we can keep this here as this is not breaking the user flow. The user just needs to register and then he can continue with the testing.

>    To be able to test with debug device models, register your development email account using the `Fast Pair Device Proposal Form`_ from Google.

Reviewer: ```suggestion
   To be able to test with debug device models, register your development email account by completing Google's `Fast Pair Device Proposal Form`_.
```

You can test the sample against the Eddystone test suite from the `Fast Pair Validator app`_.

.. note::

>    To test this section, you need to use your project in the Google Nearby Console and the Fast Pair device model that is defined in the scope of your project.

Reviewer: ```suggestion
   To test this section, use your project in the Google Nearby Console and the Fast Pair device model that is defined in the scope of your project.
```

   The default debug device model from Nordic Semiconductor cannot be used for this purpose.

>    Apart from that, you need to sign into the `Fast Pair Validator app`_ using an email address that is associated with your project in the Google Nearby Console.

Reviewer: ```suggestion
   Additionally, you must sign into the `Fast Pair Validator app`_ using an email address associated with your project in the Google Nearby Console.
```

You can test the sample against the Eddystone test suite from the `Fast Pair Validator app`_.

.. note::

>    To test this section, you need to use your project in the Google Nearby Console and the Fast Pair device model that is defined in the scope of your project.

Reviewer 2: ```
To locally perform tests from `Fast Pair Validator app`_, you need to use Fast Pair device Model ID that is defined for your project in Google Nearby Console.
You cannot use the debug device Model ID that is used by default by Fast Pair samples in the |NCS|.
...

```
